cmd/admin-server-info.go

```diff
@@ -42,9 +45,18 @@ func getLocalServerProperty(endpointServerPools EndpointServerPools, r *http.Req
 	if globalIsDistErasure {
 		addr = globalLocalNodeName
 	}
+	poolNumbers := make(map[int]struct{})
 	network := make(map[string]string)
-	for _, ep := range endpointServerPools {
+	for idx, ep := range endpointServerPools {
 		for _, endpoint := range ep.Endpoints {
+			if addr == endpoint.Host {
+				poolNumbers[idx+1] = struct{}{}
+			} else if host, err := xnet.ParseHost(addr); err == nil {
+				if host.Name == endpoint.Hostname() {
+					poolNumbers[idx+1] = struct{}{}
+				}
+			}
```

Comment:
I think another simpler way is to use endpoint.IsLocal, since this function name is called getLocalServerProperty() and returns information about the current node. It is better than parsing those addresses

Reply:
Oh yeah that is also doable